wal/decoder.go

```diff
@@ -98,6 +113,44 @@ func (d *decoder) decodeRecord(rec *walpb.Record) error {
 	return nil
 }
 
+// isTornEntry determines whether the last entry of the WAL was partially written
+// and corrupted because of a torn write.
+func (d *decoder) isTornEntry(data []byte) bool {
```

Comment:
If WAL guarantees a sync after every record this would work. Otherwise, I think WAL could tear like [A][0][A][B] in which case the WAL should truncate to before record A since there was no sync.